src/app/Caches/Create/AdvancedOptions.tsx

```diff
@@ -0,0 +1,300 @@
+    advancedOptions: AdvancedConfiguration,
+    advancedOptionsModifier: (AdvancedOptions) => void,
+}) => {
+    const { t } = useTranslation();
+
+    const [storage, setStorage] = useState<"heap" | "off_heap">(props.advancedOptions.storage || "heap");
+    const [maxSize, setMaxSize] = useState<number>(props.advancedOptions.maxSize);
+    const [maxCount, setMaxCount] = useState<number>(props.advancedOptions.maxCount);
+    const [evictionStrategy, setEvictionStrategy] = useState<string>(props.advancedOptions.evictionStrategy);
+    const [concurrencyLevel, setConcurrencyLevel] = useState<number>(props.advancedOptions.concurrencyLevel || 0);
+    const [isolationLevel, setIsolationLevel] = useState(props.advancedOptions.isolationLevel);
+    const [lockAcquisitionTimeout, setLockAcquisitionTimeout] = useState<number>(props.advancedOptions.lockAcquisitionTimeout || 0);
+    const [striping, setStriping] = useState<boolean>(props.advancedOptions.striping || true);
+
+    const [isOpenEvictionStrategy, setIsOpenEvictionStrategy] = useState(false);
+    const [isOpenIsolationLevel, setIsOpenIsolationLevel] = useState(false);
+    const [inputValueMaxSize, setInputValueMaxSize] = useState<number>(props.advancedOptions.maxSize);
+    const [inputValueMaxCount, setInputValueMaxCount] = useState<number>(props.advancedOptions.maxSize);
+
+    useEffect(() => {
+        props.advancedOptionsModifier({
+            storage: storage,
+            maxSize: maxSize,
+            maxCount: maxCount,
+            evictionStrategy: evictionStrategy,
+            concurrencyLevel: concurrencyLevel,
+            isolationLevel: isolationLevel,
+            lockAcquisitionTimeout: lockAcquisitionTimeout,
+            striping: striping,
+        });
+    }, [storage, maxSize, maxCount, evictionStrategy, concurrencyLevel, isolationLevel, lockAcquisitionTimeout, striping]);
+
+    const onToggleEvictionStrategy = () => {
+        setIsOpenEvictionStrategy(!isOpenEvictionStrategy);
+    };
+
+    const onSelectEvictionStrategy = (event, selection, isPlaceholder) => {
+        setEvictionStrategy(selection);
+        setIsOpenEvictionStrategy(false);
+        console.log('selected:', selection);
+    };
+
+    const onToggleIsolationLevel = () => {
+        setIsOpenIsolationLevel(!isOpenIsolationLevel);
+    };
+
+    const onSelectIsolationLevel = (event, selection, isPlaceholder) => {
+        setIsolationLevel(selection);
+        setIsOpenIsolationLevel(false);
+        console.log('selected:', selection);
+    };
+
+    const onToggleStriping = () => {
+        setStriping(!striping);
+    };
+
+    const handleConcurrencyLevel = (value) => {
+        setConcurrencyLevel(value);
+    }
+
+    const handleLockAcquisitionTimeout = (value) => {
+        setLockAcquisitionTimeout(value);
+    }
+
+    const onChangeMaxSize = (value, inputValue, setLocalInputValue) => {
+        let newValue;
+        if (inputValue === undefined) {
```

Comment:
extract to a utility method and add unit test

Reply:
Instead of extracting to utility function, i have created a new helper function in the same file. Would be okay?

Reply:
for  unit testing if it's in another file